### PR TITLE
[NEXUS-8429] Rip out remaining user password "reset" bits from UI/backend

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8429

### CODE REVIEW

**Comment by kellyrob99** [comment]
+1

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF192-1 :white_check_mark:

File: components/nexus-core/src/main/java/org/sonatype/nexus/internal/security/StaticSecurityConfigurationResourceImpl.groovy [hunk 1]
```diff
@@ -909,16 +909,19 @@ class StaticSecurityConfigurationResourceImpl
...
-                type: 'method',
-                name: 'User Reset Password - (delete,read)',
-                description: 'Give permission to reset any user\'s password.',
-                properties: [
-                    'method'    : 'delete,read',
-                    'permission': 'security:usersreset'
-                ]
-            ),
+
+            // NOTE: This privilege is no longer used by anything
```
Comment (adreghiciu) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/security/StaticSecurityConfigurationResourceImpl.groovy]: Why not just remove?

**Comment by adreghiciu** [comment]
+1

File: components/nexus-core/src/main/java/org/sonatype/nexus/internal/security/StaticSecurityConfigurationResourceImpl.groovy (hunk 1 above)
Comment (jdillon) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/security/StaticSecurityConfigurationResourceImpl.groovy]: I left here just to keep record of what 'id: 59' was.

I expect we'll need to overhaul the permissions at some time in the future to normalize namespace, etc and we can remove it then.

**Pull request merged**